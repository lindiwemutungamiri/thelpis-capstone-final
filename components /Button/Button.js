import React from 'react';
import { Text, Image, TouchableOpacity } from 'react-native';
import style from './style';

/**
 * The props are going to allow us to have different buttons across the entire page 
 * Creating our own custom button that we can have control over the way it looks 
 * @param {*} props 
 * @returns 
 */

export default function Button(props) {
    //the button props so that we can have a customized button wherever we want to use it 
    //if we have a text, then pass in a text
    //if we have an icon, then pass in an icon
    const { text, icon, disabled, btnStyle, onPress, btnTextStyle } = props;
    return (
        <TouchableOpacity
            activeOpacity={0.5}
            onPress={onPress}
            disabled={disabled}
            style={[style.btnContainer, btnStyle]}
        >

            {text && (
                <Text
                    style={[
                        style.btnText, style.btnTextStyle,
                        btnTextStyle,
                        { marginRight: icon ? 10 : 0 },
                    ]}
                >
                    {text}
                </Text>
            )}
            {icon && <Image source={icon} resizeModel="contain" style={style.iconStyle} />}

        </TouchableOpacity>

    );
}