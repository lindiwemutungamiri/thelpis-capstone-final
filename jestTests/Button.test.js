/**
 * Testing for the universal Button component 
 */
import { shallow } from 'enzyme';
import Button from '../components /Button/Button';

describe('UniversalButton', () => {
    it('renders correctly', () => {
      const wrapper = shallow(<Button />);
      expect(wrapper).toMatchSnapshot();

    });
  });

describe('UniversalButton', () => {
    describe('when user clicks the Button', () => {
      it ('calls correct function to save the information', () => {
        const onButtonClickMock = jest.fn();
        const wrapper = shallow(
          <Button
            onButtonClick={onButtonClickMock}
          />,
        );
        const buttonElement = wrapper.find('.Button'); 
        buttonElement.simulate('click'); 
        
        expect(onButtonClickMock).toHaveBeenCalledTimes(1); 
        expect(onButtonClickMock).toHaveBeenCalledWith(true);
      });
    });
  });