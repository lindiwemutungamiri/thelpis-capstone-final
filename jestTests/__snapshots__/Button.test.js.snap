// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button across all screens 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#0B0B45",
      "borderRadius": 10,
      "flexDirection": "row",
      "height": 98,
      "justifyContent": "center",
      "opacity": 1,
      "width": "100%",
    }
  }
/>
`;
